### DISCUSSION TITLE
Depth first traversal OrderBy example

### DISCUSSION THREAD

**Discussion by jedi1992** [EvgSkv/logica#161] [Q&A]
Hi, 
Will be great if you can add an example for Depth First traversal in a graph. If you have a tree like
a
b c (under a) 
b1 b2 ( under b)
c1 c2 ( under c)
Some query with an OrderBy that will  produce  the output  a, b, b1,b2, c, c1, c2 
Most of the graph examples dont have an OrderBy. The Victoria example with child, grandchild is almost there but it does not have an Depth first OrderBy.

Thanks in Advance.

**Comment by EvgSkv** [comment]
This is a fun puzzle, I'll do it in nearest days if noone jumps onto this thread with the solution :-)

**Reply by jedi1992** [reply]
Just to add to it , one way of solving would be come up with some level ID , sequence ID etc and then do an orderby based on it.  We will be using some form of recursive definition ( very much like the queen victoria problem)

But if you were to compare with a imperative recursive program or an equivalent loop program with a stack ( without recursion), in both the cases we can do it without labeling the data  with any IDs.  i.e the order of visiting itself is enough and we can control the order of visiting.

In Logica since we do not have the control on the order of visiting we need to add some form of extra ID to the data.  There is no other way out. Is my understanding correct ?

**Reply by EvgSkv** [reply]
@jedi1992 
Yes, that's right, at least for now we will need an id. And assigning it turns out to be trickier than I thought. Generally it looks like depth first search is an algorithm that seems to "resist" distributed computing implementation.
